netlogo-core/src/main/api/Workspace.scala

```diff
@@ -85,4 +85,12 @@ with ViewSettings with Controllable {
   def renderer: RendererInterface
   def compilerTestingMode: Boolean
   def warningMessage(message: String): Boolean
+
+  @throws(classOf[CompilerException])
+  def evaluateCommands(owner: JobOwner, source: String): Unit
+  @throws(classOf[CompilerException])
+  def evaluateCommands(owner: JobOwner, source: String, waitForCompletion: Boolean): Unit
+
+  @throws(classOf[CompilerException])
+  def evaluateReporter(owner: JobOwner, source: String): AnyRef
```

Comment:
`command` and `report` are found in the `Controllable` trait, which this trait extends. I think it *would* make sense to move all three of these methods into that trait as well so that all controlling API methods were kept in one place.